### ISSUE TITLE
Get reset time with a function

### ISSUE THREAD

**Issue by someguy9** [upstash/ratelimit-js#111]
Would love to see `ratelimit.getRemaining` also return the reset time (like limit does) or a dedicated `ratelimit.getReset` function. The issue I am having is that .limit always adds 1 to the limit even if `{ rate: 0 }` is set. So there is no way to retrieve the time at which the limit will be reset without triggering a use.

**Comment by CahidArda** [comment]
Hi @someguy9,

I think having a `getReset` method makes sense. I will work on adding it.

Using `.limit` with `rate: 0` shouldn't have reduced the rate. Which rate limiting algorithm are you using? That should be fixed aswell.

**Comment by CahidArda** [comment]
I looked into adding reset to the result of getRemaining [here](feat: add reset field to getRemaining)[feat: add reset field to getRemaining](https://github.com/upstash/ratelimit-js/commit/f2315b6c01d6efd520bfbc9ac09d9a0edaf5380e).

Here is the planned getRemaining method:

```ts
public getRemaining = async (identifier: string): Promise<{
  remaining: number;
  reset: number;
}> => {
```

It is backwards incompatible but I think it's a simple change. I will discuss it with the team.

I didn't wan't to make a seperate `getReset` method because it would need to make similar requests to redis as `getRemaining`. reset values of fixedWindow/slidingWindow algorithms can be calculated without calling redis but I feel like that reset information is not very valueable without remaining field

**Comment by CahidArda** [comment]
Hi @someguy9,

we released v2.0.0 which includes this change.
